src/cli.rs

```diff
@@ -1,102 +1,105 @@
 use std::path::PathBuf;
 
+use clap::{ArgAction, Parser};
 use defmt_decoder::DEFMT_VERSION;
 use git_version::git_version;
 use log::Level;
 use probe_rs::Probe;
-use structopt::{clap::AppSettings, StructOpt};
 
 use crate::probe;
 
 /// Successfull termination of process.
 const EXIT_SUCCESS: i32 = 0;
 
 /// A Cargo runner for microcontrollers.
-#[derive(StructOpt)]
-#[structopt(name = "probe-run", setting = AppSettings::TrailingVarArg)]
+#[derive(Parser)]
+#[command(name = "probe-run")]
 pub struct Opts {
     /// List supported chips and exit.
-    #[structopt(long)]
+    #[arg(long)]
     list_chips: bool,
 
     /// Lists all the connected probes and exit.
-    #[structopt(long)]
+    #[arg(long)]
     list_probes: bool,
 
     /// The chip to program.
-    #[structopt(long, required_unless_one(&["list-chips", "list-probes", "version"]), env = "PROBE_RUN_CHIP")]
+    #[arg(long, required = true, conflicts_with_all = HELPER, env = "PROBE_RUN_CHIP")]
     chip: Option<String>,
 
     /// Path to chip description file, in YAML format.
-    #[structopt(long)]
+    #[arg(long)]
     pub chip_description_path: Option<PathBuf>,
 
     /// The probe to use (eg. `VID:PID`, `VID:PID:Serial`, or just `Serial`).
-    #[structopt(long, env = "PROBE_RUN_PROBE")]
+    #[arg(long, env = "PROBE_RUN_PROBE")]
     pub probe: Option<String>,
 
     /// The probe clock frequency in kHz
-    #[structopt(long, env = "PROBE_RUN_SPEED")]
+    #[arg(long, env = "PROBE_RUN_SPEED")]
     pub speed: Option<u32>,
 
     /// Path to an ELF firmware file.
-    #[structopt(name = "ELF", parse(from_os_str), required_unless_one(&["list-chips", "list-probes", "version"]))]
+    #[arg(required = true, conflicts_with_all = HELPER)]
     elf: Option<PathBuf>,
 
     /// Skip writing the application binary to flash.
-    #[structopt(
+    #[arg(
         long,
-        conflicts_with = "disable-double-buffering",
+        conflicts_with = "disable_double_buffering",
         conflicts_with = "verify"
     )]
     pub no_flash: bool,
 
     /// Connect to device when NRST is pressed.
-    #[structopt(long)]
+    #[arg(long)]
     pub connect_under_reset: bool,
 
     /// Enable more verbose output.
-    #[structopt(short, long, parse(from_occurrences))]
-    pub verbose: u32,
+    #[arg(short, long, action = ArgAction::Count)]
+    pub verbose: u8,
 
     /// Prints version information
-    #[structopt(short = "V", long)]
+    #[arg(short = 'V', long)]
```

Comment:
Did not notice before that 'V' != 'v' !

Reply:
Yes. `-v` is `--verbose` and `-V` is `--version`.